DependencyInjection/Configuration.php

```diff
@@ -564,4 +517,102 @@ private function validateAuthenticationType(array $expected, array $actual, $aut
             implode(', ', $actual)
         ));
     }
+
+    /**
+     * Create configuration for cache plugin.
+     *
+     * @return NodeDefinition Definition for the cache node in the plugins list.
+     */
+    private function createCachePluginNode()
+    {
+        $builder = new TreeBuilder();
+
+        $config = $builder->root('config');
+        $config
+            ->fixXmlConfig('method')
+            ->fixXmlConfig('respect_response_cache_directive')
+            ->addDefaultsIfNotSet()
+            ->validate()
+                ->ifTrue(function ($config) {
+                    // Cannot set both respect_cache_headers and respect_response_cache_directives
+                    return isset($config['respect_cache_headers'], $config['respect_response_cache_directives']);
+                })
+                ->thenInvalid('You can\'t provide config option "respect_cache_headers" and "respect_response_cache_directives" simultaniously. Use "respect_response_cache_directives" instead.')
+            ->end()
+            ->children()
+                ->scalarNode('cache_key_generator')
+                    ->info('A class implementing '.CacheKeyGenerator::class.' to generate a PSR-6 cache key')
```

Comment:
actually the cache plugin phpdoc is confusing on this one. if this option is set, it must contain an instance of CacheKeyGenerator. in symfony config that means it would be a service id.